### PR TITLE
chore: modernize development

### LINKED ISSUE #12: Modernize development / testing
- Drop support for Python2.x
- Add support for "modern" Python3 versions (>= 3.7)
- Convert testing to use tox / Github Actions
- Require that tests pass for merging
~ - (Maybe) add at least minimal Sphinx docs? ~

### PR DESCRIPTION
**Pull request by tseaver**
Closes #12.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.eggs/
```
Comment (dataflake) [inline .gitignore:1]: Here's the default template we use:

https://github.com/zopefoundation/meta/blob/master/config/default/gitignore.j2

File: setup.py
Comment (dataflake) [inline setup.py:1]: This is missing a `python_requires` in the `setup` function arguments. You could also update the URLs to use `https` if that is supported.

**Review by dataflake — APPROVED** [review]
We use a set of scripts to generate and maintain many of those package files, see https://github.com/zopefoundation/meta/tree/master/config

**Comment by malthe** [comment]
@dataflake there are some open suggestions here, did you want to get these current changes merged in or wait until we decide on these open suggestions?

**Pull request merged**